conda_smithy/templates/run_docker_build.tmpl

```diff
@@ -60,7 +60,11 @@ conda install --yes --quiet conda-forge-build-setup
 {{ build_setup }}{% endif -%}
 
 {%- block build %}
+{%- if 'ccache-toolchain' in ' '.join(package['meta']['requirements']['build']) %}
+conda build --no-build-id /recipe_root --quiet || exit 1
```

Comment:
Please leave a comment explaining why ```--no-build-id``` is being used. Something like:

```
ccache uses the compiler command line to generate the cache keys, so we can't have a different path for every build.
ccache provides CCACHE_BASEDIR variable to solve this but it's better to avoid setting it as it has some side effects.
```